subsys/pm/power.c

```diff
@@ -122,29 +122,31 @@ static int pm_suspend_devices(void)
 
 	for (const struct device *dev = devs + devc - 1; dev >= devs; dev--) {
 		int ret;
+		struct pm_device *pm = dev->pm;
 
 		/*
-		 * ignore busy devices, wake up source and devices with
-		 * runtime PM enabled.
+		 * skip suspension if device:
+		 * - does not support PM
+		 * - is already suspended
+		 * - is busy
+		 * - is a wake up source
+		 * - has runtime PM enabled
 		 */
-		if (pm_device_is_busy(dev) ||
+		if ((pm == NULL) || (pm->state == PM_DEVICE_STATE_SUSPENDED) ||
```

Comment:
we still have `PM_DEVICE_STATE_OFF` and that has to be accounted here.

Reply:
added back